kernel/data-pipeline/scenario/cdc/client/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/client/CDCClient.java

```diff
@@ -93,14 +82,174 @@ protected void initChannel(final NioSocketChannel channel) {
+        }
+        if (parameter.getPort() <= 0) {
+            throw new IllegalArgumentException("The port must be greater than 0");
+        }
+    }
+    
+    /**
+     * Check channel is active.
+     *
+     * @return true if channel is active
+     */
+    public boolean isActive() {
+        return channel.isActive();
+    }
+    
+    /**
+     * Await channel close.
+     *
+     * @throws InterruptedException interrupted exception
+     */
+    public void await() throws InterruptedException {
+        channel.closeFuture().sync();
+    }
+    
+    /**
+     * Login.
+     *
+     * @param parameter parameter
+     * @throws IllegalStateException     the channel is not active
+     * @throws IllegalArgumentException  the user is illegal
+     * @throws GetResultTimeoutException get result timeout
+     */
+    public synchronized void login(final CDCLoginParameter parameter) {
+        if (null == channel || !channel.isActive()) {
+            throw new IllegalStateException("The channel is not active");
+        }
+        ClientConnectionContext connectionContext = channel.attr(ClientConnectionContext.CONTEXT_KEY).get();
+        if (ClientConnectionStatus.LOGGED_IN == connectionContext.getStatus().get()) {
+            throw new IllegalStateException("The client is already logged in");
+        }
+        LoginRequestBody loginRequestBody = LoginRequestBody.newBuilder().setType(LoginType.BASIC).setBasicBody(BasicBody.newBuilder().setUsername(parameter.getUsername())
+                .setPassword(Hashing.sha256().hashBytes(parameter.getPassword().getBytes()).toString().toUpperCase()).build()).build();
+        String requestId = RequestIdUtils.generateRequestId();
+        CDCRequest data = CDCRequest.newBuilder().setType(Type.LOGIN).setVersion(1).setRequestId(requestId).setLoginRequestBody(loginRequestBody).build();
+        ResponseFuture responseFuture = new ResponseFuture();
+        connectionContext.getResponseFutureMap().put(requestId, responseFuture);
+        channel.writeAndFlush(data);
         try {
-            ChannelFuture future = bootstrap.connect(address, port).sync();
-            future.channel().closeFuture().sync();
+            getResultFromResponseFuture(responseFuture);
+            log.info("Login success, username: {}", parameter.getUsername());
         } finally {
+            connectionContext.getResponseFutureMap().remove(requestId);
+        }
+    }
+    
+    private Object getResultFromResponseFuture(final ResponseFuture responseFuture) {
+        boolean receivedResult = responseFuture.waitResponse(config.getTimeoutMills());
+        if (!receivedResult) {
+            throw new GetResultTimeoutException("Get result timeout");
+        }
+        if (!Strings.isNullOrEmpty(responseFuture.getErrorMessage())) {
+            throw new IllegalArgumentException(String.format("Get Response failed, reason: %s", responseFuture.getErrorMessage()));
+        }
+        return responseFuture.getResult();
```

Comment:
1, Seems check error message should be before check received or not.

2, Could we merge this block of code into `waitResponse`?

3, Is there underlying timeout exception to replace `GetResultTimeoutException`?

4, Could we use another exception to replace `IllegalArgumentException`? Looks it's not suitable.

Reply:
3. I did not find a more suitable , `java.util.concurrent.TimeoutException` is not a RuntimeException,